fineract-provider/src/test/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImplTest.java

```diff
@@ -259,4 +313,171 @@ public void chargeOff_forReversedTransaction_shouldRun() {
+
+            when(loan.isMultiDisburmentLoan()).thenReturn(false);
+            when(loan.isUpfrontAccrualAccountingEnabledOnLoanProduct()).thenReturn(true);
+            when(loan.getSummary().getTotalInterestCharged()).thenReturn(BigDecimal.TEN);
+
+            try (MockedStatic<TemporaryConfigurationServiceContainer> temporaryConfigServiceMock = mockStatic(TemporaryConfigurationServiceContainer.class)) {
+
+                temporaryConfigServiceMock.when(TemporaryConfigurationServiceContainer::isExternalIdAutoGenerationEnabled)
+                        .thenReturn(false);
+
+                loanWritePlatformService.disburseLoan(LOAN_ID, command, false, false);
+
+                verify(loan, times(2)).addLoanTransaction(argThat(tx -> tx.isDisbursement() || tx.isAccrual()));
+            }
+
+        }
+
+        @Test
+        public void shouldApplyAccrualInterestWhenMultiDisbursementFirstDisbursementAndUpfrontAccrual() {
+
+            when(loan.isMultiDisburmentLoan()).thenReturn(true);
+            when(loan.getDisbursedLoanDisbursementDetails()).thenReturn(List.of(mock(LoanDisbursementDetails.class)));
+            when(loan.isUpfrontAccrualAccountingEnabledOnLoanProduct()).thenReturn(true);
+            when(loan.getSummary().getTotalInterestCharged()).thenReturn(BigDecimal.TEN);
+
+            try (MockedStatic<TemporaryConfigurationServiceContainer> temporaryConfigServiceMock = mockStatic(TemporaryConfigurationServiceContainer.class)) {
+
+                temporaryConfigServiceMock.when(TemporaryConfigurationServiceContainer::isExternalIdAutoGenerationEnabled)
+                        .thenReturn(false);
+
+                loanWritePlatformService.disburseLoan(LOAN_ID, command, false, false);
+
+                verify(loan, times(2)).addLoanTransaction(argThat(tx -> tx.isDisbursement() || tx.isAccrual()));
+            }
+
+        }
+
+        @Test
+        public void shouldApplyAccrualInterestWhenCashAccountingIsEnabled() {
+
+            when(loan.isMultiDisburmentLoan()).thenReturn(false);
+            when(loan.isUpfrontAccrualAccountingEnabledOnLoanProduct()).thenReturn(false);
+            when(loan.isCashBasedAccountingEnabledOnLoanProduct()).thenReturn(true);
+            when(fineractProperties.getLoan().isAllowCashAndNonCashAccrual()).thenReturn(true);
+            when(loan.getSummary().getTotalInterestCharged()).thenReturn(BigDecimal.TEN);
+
+            try (MockedStatic<TemporaryConfigurationServiceContainer> temporaryConfigServiceMock = mockStatic(TemporaryConfigurationServiceContainer.class)) {
+
+                temporaryConfigServiceMock.when(TemporaryConfigurationServiceContainer::isExternalIdAutoGenerationEnabled)
+                        .thenReturn(false);
+                loanWritePlatformService.disburseLoan(LOAN_ID, command, false, false);
+
+                verify(loan, times(2)).addLoanTransaction(argThat(tx -> tx.isDisbursement() || tx.isAccrual()));
```

Comment:
Little weird assertion this. Can we do it as different verifications?
one for disbursement txn and one for accrual